problems/arraysWithLoops.js

```diff
@@ -24,50 +39,117 @@ function shoutWhileLoop() {}
  * @param {Array} - ["I", "am", "a", "happy", "array"]
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
-
-function shoutForOfLoop() {}
+// for (variable of iterable) {
+//   statement
+// }
+
+//const array1 = ['a', 'b', 'c'];
+// for (const element of array1) {
+//   console.log(element);
+// }
+// expected output: "a"
+// expected output: "b"
+// expected output: "c"
+
+//function shoutForOfLoop() {}
+const shoutForOfLoop = (array) => {
+  let newArr = [];
+  for (const element of array) {
+    newArr.push(element + "!")
+  }
+  return newArr;
+}
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
-
-function sumArray() {}
+//function sumArray() {}
+const sumArray = (nums) => {
+  let sum = 0;
+  for (const element of nums){
+   sum += element;
+  }
+return sum;
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
-
-function oddArray() {}
+//function oddArray() {}
+const oddArray = (nums) => {
+  let arrOfOdds = [];
+  for (const element of nums){
+    if (element % 2 === 1) {
+      arrOfOdds.push(element);
+    }
+  }
+return arrOfOdds;
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
-
-function evenArray() {}
+//function evenArray() {}
+const evenArray = (nums) => {
+  let arrOfEvens = [];
+  for (const element of nums){
+    if (element % 2 === 0){
+      arrOfEvens.push(element)
+    }
+  }
+  return arrOfEvens;
+}
 
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
-
-function findSmallest() {}
+//function findSmallest() {}
+const findSmallest = (nums) => {
+  let smallest = nums[0];
+  for (const element of nums){
+    if(element <= smallest) { 
+      smallest = element;
+    }
+  }
+    return smallest;
+}
 
 /**
  * Takes in array and returns the second smallest number in the array
  * You may NOT use sort
  * @param {number[]} nums
  * @returns {number} second smallest value
  */
-
-function findSecondSmallest() {}
-
+//function findSecondSmallest() {}
+const findSecondSmallest = (nums) =>{
+  let smallest = nums[0];
+  let secondSmallest = nums[1];
```

Comment:
It's better to set these to Infinist than the first variable because if num[0] is 50 and num[1] is 1 then num[1] is actually our smallest